packages/sync-service/lib/electric/shapes/api.ex

```diff
@@ -591,6 +597,22 @@ defmodule Electric.Shapes.Api do
     }
   end
 
+  defp do_serve_shape_log(%Request{new_changes_ref: ref} = request)
+       when is_out_of_bounds(request) do
+    # treat out of bounds requests like live requests with a
+    # shorter timeout before failing them, as if the client happened
+    # to be slightly ahead because of a restart or handover the
+    # offset they have seen should show up shortly, otherwise we
+    # assume it is an actually invalid out of bounds request
+    Process.send_after(
+      self(),
+      {ref, :out_of_bounds_timeout},
+      floor(request.api.long_poll_timeout * 0.5)
```

Comment:
```suggestion
      div(request.api.long_poll_timeout, 2)
```
more idiomatic this way, IMO.